packages/suite/src/middlewares/wallet/coinjoinMiddleware.ts

```diff
@@ -164,15 +88,77 @@ export const coinjoinMiddleware =
             }
         }
 
+        // Pause coinjoin session when device disconnects.
+        // This is not treated a temporary interruption with automatic restore because the user probably disconnects the device willingly.
+        if (action.type === DEVICE.DISCONNECT && action.payload.id) {
+            api.dispatch(coinjoinAccountActions.pauseCoinjoinSessionByDeviceId(action.payload.id));
+        }
+
+        // Pause/restore coinjoin session when Suite goes offline/online.
+        // This is just UX improvement as the session could not continue offline anyway.
+        if (action.type === SUITE.ONLINE_STATUS) {
+            if (action.payload === false) {
+                api.dispatch(coinjoinAccountActions.pauseInterruptAllCoinjoinSessions());
+            } else if (action.payload === true) {
+                api.dispatch(coinjoinAccountActions.restoreInterruptedCoinjoinSessions());
+            }
+        }
+
+        // Pause/restore coinjoin session based on Tor status.
+        // Continuing coinjoin would be a privacy risk.
         if (action.type === SUITE.TOR_STATUS) {
             if (['Disabling', 'Disabled', 'Error'].includes(action.payload)) {
                 api.dispatch(coinjoinAccountActions.pauseInterruptAllCoinjoinSessions());
+            } else if (action.payload === 'Enabled') {
+                api.dispatch(coinjoinAccountActions.restoreInterruptedCoinjoinSessions());
```

Comment:
Check for other blockers here too like on line 133?

Reply:
The check is done inside `restoreInterruptedCoinjoinSessions` for each session. I included it in the function as it would otherwise repeat.